### ISSUE TITLE
Binding "raise or lower window" to a key or mouse button

### ISSUE THREAD

**Issue by tilo** [sbmpost/AutoRaise#133]
TL;DR: 

* is it possible to separate the focus-follows-mouse behavior from raising the window?
* can raise-or-lower-window be bound to a key or mouse button?

Background:

On UNIX window managers focus-follows-mouse and raise-or-lower-window are two separate options.

With that, you can point your mouse at a window that is partially hidden, e.g. a terminal, and into type into it. This means you don't have to constantly re-arrange your windows just to interact. 

You can raise or lower the window under the mouse pointer by hitting one key. 
If the window was on top, it goes to the very bottom. If the window was partially covered, it goes to the top. That means you might have to hit that key multiple times to get to a window in the middle. All that without having to change the layout of your windows or to see a gazillion windows in Mission Control view.

**Comment by yarub123** [comment]
I thought I figured out what you meant, but the more I read, the more confused I got. I have absolutely no idea what you mean. 

Are you simply talking about the window gaining auto focus and rising to the top once you type into it? 

For example if I have TextEdit behind Firefox, and FF is on top, TE is in the back, I can hover over with my mouse, start typing and the TE window jumps to front?

If that was the case then yea would be cool to have the option where the window you start typing in, if it is in the back, that it would auto-gain-focus.

**Comment by sbmpost** [comment]
@tilo 

> With that, you can point your mouse at a window that is partially hidden, e.g. a terminal, and into type into it. This means you don't have to constantly re-arrange your windows just to interact.

This is possible by setting:

    -delay 0 -focusDelay 1

In the UI version you would drag the raise delay slider completely to the left to disable raising and have focus only.

**Comment by tkunze** [comment]
long-time `-delay 0 -focusDelay 1` user here and wanted to add my vote for the "send window to back" option mentioned here, just like in X eons ago. One key to send to back, one to raise to top (I know we can already click a window in the back to bring it to the front; that key would be merely for symmetry reasons)
